File: parse_ts.js
const Parser = require('tree-sitter');
const TypeScript = require('tree-sitter-typescript').typescript;

const parser = new Parser();
parser.setLanguage(TypeScript);

const sourceCode = 'export namespace Auth { export function login() {} }';
const tree = parser.parse(sourceCode);

console.log(tree.rootNode.toString());

File: parse_ts.rs
use tree_sitter::Parser;

fn main() {
    let mut parser = Parser::new();
    parser.set_language(tree_sitter_typescript::LANGUAGE_TYPESCRIPT.into()).unwrap();

    let source = "export declare namespace Auth { function login(): void; }";
    let tree = parser.parse(source, None).unwrap();
    println!("{}", tree.root_node().to_sexp());
}
